structure/stack/stack_array.go

```diff
@@ -0,0 +1,89 @@
+// Stack implementation using Array
+// Stack is a linear data structure in which the push and pop operations occur only at one end of the structure, referred to as the top of the stack.
+// The order in which an element added to or removed from a stack is described as last in, first out (LIFO).
+// Details:
+// 	Stack Data Structure : https://www.geeksforgeeks.org/stack-data-structure-introduction-program/
+// 	Stack (abstract data type) : https://en.wikipedia.org/wiki/Stack_(abstract_data_type)
+
+package stack
+
+import "errors"
+
+var ErrStackEmpty = errors.New("stack is empty")
+
+/*
+The methods can also be implemented directly on the slice.
+```
+type Array[T any] []T
+```
+However, this exposes the underlaying storage (slice) outside the package.
+A struct is used instead, so that the underlying storage is not accessible outside the package.
+*/
+
+// Array is an implementation of stack with slice as underlying storage.
+// ```
+// stack := stack.NewArray[int]()
+// ```
```

Comment:
```suggestion
// Array is an implementation of stack with slice as underlying storage.
// ```
// stack := stack.NewArray[int]()
// ```
// 
// Note that the type `Array` could also be implemented directly using a slice.
// ```
// type Array[T any] []T
// ```
// However, this exposes the underlaying storage (slice) outside the package.
// A struct is used instead, so that the underlying storage is not accessible
// outside the package.
```

Reply:
@tjgurwara99 Updated with all changes. Interface is specified in `stack/stack.go` file.